### ISSUE TITLE
[镜像请求] 增加 CachyOS 镜像

### ISSUE THREAD

**Issue by heheda123123** [tuna/issues#1901]
### 先决条件 (Prerequisites)

- [X] 我已确认这个镜像源从未在 [其他 issues](https://github.com/tuna/issues) 中讨论过。
I am sure that this repo has NEVER been discussed in [other issues](https://github.com/tuna/issues).

- [X] 我已确认这个镜像源没有我选择的镜像站上线。
I am sure that this repo has NOT currently be provided by the sites I select.

- [X] 我已仔细阅读 [请求新镜像前必读（试行）](https://github.com/tuna/issues/issues/741)。
I have CAREFULLY read the [mirroring rules](https://github.com/tuna/issues/issues/741).

### 项目名称与简介 (Project introduction)

CachyOS

官网地址 https://cachyos.org/
镜像地址 https://aur.cachyos.org/

archlinux的衍生版，主要特点如下
1 软件默认启用了 LTO 和 x86-64-v3 优化
2 默认启用 linux-cachyos 内核

### 上游地址与镜像方法 (How to mirror)

自带的几个镜像文件
![image](https://github.com/tuna/issues/assets/97490782/8b02e0d3-8f49-4557-960c-c931d67b898f)
```
❯ cat /etc/pacman.d/cachyos-mirrorlist 
######################################################
####                                              ####
####        CachyOS Repository Mirrorlist         ####
####                                              ####
######################################################
#### Entry in file /etc/pacman.conf:
###     [cachyos]
###     Include = /etc/pacman.d/cachyos-mirrorlist
######################################################
## Germany
Server = https://aur.cachyos.org/repo/$arch/$repo
Server = https://mirror.cachyos.org/repo/$arch/$repo
## Hungary much thanks to ArchanfelHUN!
Server = https://hun.cachyos.org/repo/$arch/$repo
## South Korea much thanks to silent_heigou!
Server = https://kr.cachyos.org/repo/$arch/$repo
## USA Mirror much thanks to corpdecker!
Server = https://us.cachyos.org/repo/$arch/$repo
## Signapore Mirror much thanks to innoix
Server = https://sg.mirror.cx/cachyos/repo/$arch/$repo
## USA CDN Mirror much thanks to soulharsh007!
Server = https://us.soulharsh007.dev/repo/$arch/$repo
## India Mirror much thanks to https://github.com/albonycal
Server = https://mirror.albony.xyz/cachylinux/repo/$arch/$repo
## France Mirror much thanks to Antoine Viallon (aviallon)
Server = https://mirror.lesviallon.fr/cachy/repo/$arch/$repo
## Russia Mirror
Server = https://mirror.truenetwork.ru/cachy/repo/$arch/$repo
## Norway Mirror much thanks to QuadFeed
Server = https://mirror.fast0ne.com/repo/$arch/$repo
## Norway Mirror much thanks to innoix
Server = https://no.mirror.cx/cachyos/repo/$arch/$repo
```
```
❯ cat /etc/pacman.d/cachyos-v3-mirrorlist
######################################################
####                                              ####
####        CachyOS Repository Mirrorlist         ####
####                                              ####
######################################################
#### Entry in file /etc/pacman.conf:
###     [cachyos-v3]
###     Include = /etc/pacman.d/cachyos-v3-mirrorlist
######################################################
## Germany
Server = https://mirror.cachyos.org/repo/$arch_v3/$repo
Server = https://aur.cachyos.org/repo/$arch_v3/$repo
## Hungary much thanks to to ArchanfelHUN!
Server = https://hun.cachyos.org/repo/$arch_v3/$repo
## Singapore much thanks to SM9!
Server = https://sg.cachyos.org/repo/$arch_v3/$repo
## South Korea much thanks to silent_heigou!
Server = https://kr.cachyos.org/repo/$arch_v3/$repo
## USA Mirror much thanks to corpdecker!
Server = https://us.cachyos.org/repo/$arch_v3/$repo
## India Mirror much thanks to https://github.com/albonycal
Server = https://mirror.albony.xyz/cachylinux/repo/$arch_v3/$repo
## France Mirror much thanks to Antoine Viallon (aviallon)
Server = https://mirror.lesviallon.fr/cachy/repo/$arch_v3/$repo
## Russia Mirror
Server = https://mirror.truenetwork.ru/cachy/repo/$arch_v3/$repo
```
```
❯ cat /etc/pacman.d/cachyos-v4-mirrorlist
######################################################
####                                              ####
####        CachyOS Repository Mirrorlist         ####
####                                              ####
######################################################
#### Entry in file /etc/pacman.conf:
###     [cachyos-v4]
###     Include = /etc/pacman.d/cachyos-v4-mirrorlist
######################################################
## Germany
Server = https://mirror.cachyos.org/repo/$arch_v4/$repo
Server = https://aur.cachyos.org/repo/$arch_v4/$repo
## Hungary much thanks to to ArchanfelHUN!
Server = https://hun.cachyos.org/repo/$arch_v4/$repo
## South Korea much thanks to silent_heigou!
Server = https://kr.cachyos.org/repo/$arch_v4/$repo
## Singapore much thanks to SM9!
Server = https://sg.cachyos.org/repo/$arch_v4/$repo
## USA Mirror much thanks to corpdecker!
Server = https://us.cachyos.org/repo/$arch_v4/$repo
## India Mirror much thanks to https://github.com/albonycal
Server = https://mirror.albony.xyz/cachylinux/repo/$arch_v4/$repo
## France Mirror much thanks to Antoine Viallon (aviallon)
Server = https://mirror.lesviallon.fr/cachy/repo/$arch_v4/$repo
## Russia Mirror
Server = https://mirror.truenetwork.ru/cachy/repo/$arch_v4/$repo

```

### 期望上线的镜像站 (Expected mirroring sites)

- [X] TUNA - https://mirrors.tuna.tsinghua.edu.cn
- [X] BFSU - https://mirrors.bfsu.edu.cn

**Comment by ptr1337** [comment]
Hi,

We are searching currently for a mirror hosting for CachyOS in China, since user seem to have sometimes problem to connect behind mirrors/buckets behind cloudflare.

Feel free to get in contact via:
admin@cachyos.org

Thanks in advance!

**Comment by Harry-Chen** [comment]
@ptr1337 Thanks for contacting us. I have some questions:

* Do you provide rsync upstream?
* What is the approximate size of current repo, and at what speed would it grow?

**Comment by ptr1337** [comment]
> @ptr1337 Thanks for contacting us. I have some questions:
> 
> * Do you provide rsync upstream?
> * What is the approximate size of current repo, and at what speed would it grow?

Hi,

1. Yes, we do. I would contact you via E-Mail and send you the rsync. The pull intervall should be around 30 to 60 minutes, if possible.
2. The current size (including ISOs) has around 100GB without debug packages (these are only on one server seperately saved and running debuginfod there). It is not planned to get over 150GB. We might add another march, but this is still in discussion.

**Comment by heheda123123** [comment]
Any progress?

**Comment by kailanyue** [comment]
> Any progress?

I am also using this OS, Keep following

**Comment by ma0shu** [comment]
Any progress Now?

**Comment by dsanke** [comment]
Any progress Now?

**Comment by ptr1337** [comment]
Still waiting for E-Mail :( If anyone has another direct contact, let me know

**Comment by heheda123123** [comment]
> Still waiting for E-Mail :( If anyone has another direct contact, let me know

maybe this email `support@tuna.tsinghua.edu.cn`

**Comment by ptr1337** [comment]
> Please use [mirroradmin@tuna.tsinghua.edu.cn](mailto:mirroradmin@tuna.tsinghua.edu.cn) to contact us. Due to insufficient space, we have to postpone some mirror requests.
> 
> Now that we have some free space due to centos 7 and debian 10 being removed from upstream, I can start syncing once I get the rsync endpoint from upstream.

I have sent you the details about the rsync link, size and sync interval. 
Let me know, when its done, then I will add it to our mirrorlist. New Installation will then automatically ranked and existing once can run "cachyos-rate-mirrors"

**Comment by Harry-Chen** [comment]
> I have sent you the details about the rsync link, size and sync interval. Let me know, when its done, then I will add it to our mirrorlist. New Installation will then automatically ranked and existing once can run "cachyos-rate-mirrors"

Thanks for your mail. I have now started the initial synchronization. The speed is about ~50 Mbps, with which the process might take roughly one day or longer. I will let you know once it is finished.

I plan to use <https://mirrors.tuna.tsinghua.edu.cn/cachyos/> (note: currently contains **incomplete content**!) to serve the files. Is that okay?

**Comment by ptr1337** [comment]
> > I have sent you the details about the rsync link, size and sync interval. Let me know, when its done, then I will add it to our mirrorlist. New Installation will then automatically ranked and existing once can run "cachyos-rate-mirrors"
> 
> Thanks for your mail. I have now started the initial synchronization. The speed is about ~50 Mbps, with which the process might take roughly one day or longer. I will let you know once it is finished.
> 
> I plan to use https://mirrors.tuna.tsinghua.edu.cn/cachyos/ (note: currently contains **incomplete content**!) to serve the files. Is that okay?

Yes, that sounds good to me! I will add it as soon the sync is complete. Thanks for hosting that!

**Comment by Harry-Chen** [comment]
The mirror is now up and running. We also would request volunteers from CachyOS society to contribute to:

* https://github.com/mirrorz-org/genisolist
* https://github.com/mirrorz-org/mirrorz-docs (Note: all content is in Chinese)

So that users could better leverage the mirror.

**Comment by ptr1337** [comment]
@Harry-Chen Thanks for completing the sync!

I have added it to our mirrorlists. Existing user can rerank the mirrors with "cachyos-rate-mirrors".

I will look into above mentioned repositories and contribute the required changes.

**Comment by Harry-Chen** [comment]
We spot a problem at https://wiki.cachyos.org/policy/repository_policy/:

```
2. Authorized Users
The CachyOS repository is exclusively available for the following users:

CachyOS Users ArchLinux Users

3. Unauthorized Use
Use of the CachyOS repository by any other Linux distributions, including other Arch-based distributions, is strictly prohibited. This includes, but is not limited to:

Manjaro EndeavourOS ArcoLinux Parabola Any other Linux distribution not explicitly mentioned in the “Authorized Users” section
```

We are using Debian, and the policy obviously prohibits our "use". Also, we could not guarantee that all our users would comply with it, which constructs a high legal risk of violation.

I have stopped synchronization now, and are planning a removal. @yaoge123 You might also want to remove it from NJU mirrors.

**Comment by ptr1337** [comment]
> We spot a problem at [wiki.cachyos.org/policy/repository_policy](https://wiki.cachyos.org/policy/repository_policy/):
> 
> ```
> 2. Authorized Users
> The CachyOS repository is exclusively available for the following users:
> 
> CachyOS Users ArchLinux Users
> 
> 3. Unauthorized Use
> Use of the CachyOS repository by any other Linux distributions, including other Arch-based distributions, is strictly prohibited. This includes, but is not limited to:
> 
> Manjaro EndeavourOS ArcoLinux Parabola Any other Linux distribution not explicitly mentioned in the “Authorized Users” section
> ```
> 
> We are using Debian, and the policy obviously prohibits our "use". Also, we could not guarantee that all our users would comply with it, which constructs a high legal risk of violation.
> 
> I have stopped synchronization now, and are planning a removal. @yaoge123 You might also want to remove it from NJU mirrors.

Hi Harry,

Mirroring the repository is not a problem and you can proceed keeping the mirror up.

**Comment by him007** [comment]
So the problem is that the repo policy not clearly enough for end users, distribution maintainers and mirror maintainers.

**Comment by ptr1337** [comment]
https://github.com/CachyOS/wiki/compare/0716079fa171...a194d678df26

I have added a note to clarify it for mirrors.

**Comment by kailanyue** [comment]
I am a cachyos user from China, and I really enjoy using cachyos. Although I can use cachyos without relying on mirror sites, I am concerned about the current situation. I hope that everything can be resolved perfectly.
😮😕

**Comment by TechCiel** [comment]
> This is included, because distributions have started to use the repository, and started to blame us, if the users at their distribution had issues, and therefore we simply disallowed to usage of it.

I totally understand your concern, but it still makes no sense for an open-source to *prohibit* somebody from using, since it is and should always be public knowledge. Almost every free license waives any liability or warranty by their nature.

You may reiterate that *NO WARRANTY* and *STRONGLY DISCOURAGE* them from using, but I'd suggest your project refrain from prohibiting those. It a legally binding statement contradicting you license, and shows a figure of tyrant instead of community.

**Comment by ljlofficial** [comment]
@Harry-Chen 貌似它们的Wiki已经进行了更改，请酌情启动镜像同步。

**Comment by rtransformation** [comment]
I heard one user of Bilibili said that he used the Korea mirror is fast enough for him, I do not know if this is true, since I haven't started using CachyOS yet, but I've decided to try installing it when I go home for the Chinese New Year holiday.

我听一个B站用户说他用韩国的一个镜像速度很快，不知道是不是真的，我还没开始用CachyOS，决定过年放假回家安一个试试。

**Comment by ptr1337** [comment]
> I heard one user of Bilibili said that the Korea mirror is fast enough for him, I do not know if this is true, since I haven't started using CachyOS yet, but I've decided to try installing it when I go home for the Chinese New Year holiday.
> 
> 我听一个B站用户说他用韩国的镜像速度很快，不知道是不是真的，我还没开始用CachyOS，决定过年放假回家安一个试试。

We have also a world wide cdn, which has several 27 data centeres in asia. One of it in hong kong.

**Comment by shankerwangmiao** [comment]
I fully understand that installing CachyOS packages on unsupported distributions will cause unexpected problems and possible user questions or complains on this may add burden to the maintainers. However, to **prohibit** users from doing so in the terms and conditions or in the license is not acceptable, since it may violate the original licenses of the packaged softwares, which may require any derived work also following the same license. 

As a result, we cannot provide mirror service for CachyOS packages, and we have removed it from our servers.

**Comment by SoulHarsh007** [comment]
:wave: I am not sure if this was mentioned, but we disallow distributions to ship with our repository out of the box, the users from other distributions are free to use our repositories. The distributions may not ship our repositories as a part of their own distribution. Does it still violate your policy?

**Comment by rostam300** [comment]
Any progress on this ? I am still waiting for you to add cachyos , with much love ❤️❤️❤️
